Windows cmake failed compile for rocm

https://github.com/ggerganov/llama.cpp/pull/6797#issuecomment-2072608956

```
[1/81] Linking CXX static library llama.lib
[2/81] Linking CXX static library ggml_static.lib
[3/81] Linking CXX static library common\common.lib
[4/81] Linking CXX executable bin\test-c.exe
[5/81] Linking CXX executable bin\test-tokenizer-0.exe
[6/81] Linking CXX executable bin\test-tokenizer-1-bpe.exe
[7/81] Linking CXX executable bin\test-tokenizer-1-spm.exe
[8/81] Linking CXX executable bin\test-quantize-fns.exe
[9/81] Linking CXX executable bin\test-quantize-perf.exe
[10/81] Linking CXX executable bin\test-sampling.exe
[11/81] Linking CXX executable bin\test-chat-template.exe
[12/81] Linking CXX executable bin\test-grammar-parser.exe
[13/81] Linking CXX executable bin\test-llama-grammar.exe
[14/81] Linking CXX executable bin\test-grammar-integration.exe
[15/81] Linking CXX executable bin\test-grad0.exe
[16/81] Building CXX object examples/llava/CMakeFiles/llava-cli.dir/llava-cli.cpp.obj
FAILED: examples/llava/CMakeFiles/llava-cli.dir/llava-cli.cpp.obj 
ccache C:\PROGRA~1\AMD\ROCm\5.7\bin\CLANG_~1.EXE -DGGML_CUDA_DMMV_X=32 -DGGML_CUDA_MMV_Y=1 -DGGML_SCHED_MAX_COPIES=4 -DGGML_USE_CUDA -DGGML_USE_HIPBLAS -DGGML_USE_LLAMAFILE -DK_QUANTS_PER_ITERATION=2 -D_CRT_SECURE_NO_WARNINGS -D_XOPEN_SOURCE=600 -D__HIP_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -IC:/Users/andre/Desktop/development/llama.cpp/examples -IC:/Users/andre/Desktop/development/llama.cpp/common/. -IC:/Users/andre/Desktop/development/llama.cpp/. -IC:/Users/andre/Desktop/development/llama.cpp/examples/llava/. -IC:/Users/andre/Desktop/development/llama.cpp/examples/llava/../.. -IC:/Users/andre/Desktop/development/llama.cpp/examples/llava/../../common -isystem "C:/Program Files/AMD/ROCm/5.7/include" -O3 -DNDEBUG -D_DLL -D_MT -Xclang --dependent-lib=msvcrt -std=gnu++14 -Wmissing-declarations -Wmissing-noreturn -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wunreachable-code-break -Wunreachable-code-return -Wmissing-prototypes -Wextra-semi -march=native -MD -MT examples/llava/CMakeFiles/llava-cli.dir/llava-cli.cpp.obj -MF examples\llava\CMakeFiles\llava-cli.dir\llava-cli.cpp.obj.d -o examples/llava/CMakeFiles/llava-cli.dir/llava-cli.cpp.obj -c C:/Users/andre/Desktop/development/llama.cpp/examples/llava/llava-cli.cpp
In file included from C:/Users/andre/Desktop/development/llama.cpp/examples/llava/llava-cli.cpp:2:
C:/Users/andre/Desktop/development/llama.cpp/common/.\log.h:300:48: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
    #define LOG(str, ...) LOG_IMPL("%s" str, "", ##__VA_ARGS__, "")
                                               ^
C:/Users/andre/Desktop/development/llama.cpp/common/.\log.h:251:106: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
            fprintf(LOG_TARGET, LOG_TIMESTAMP_FMT LOG_FLF_FMT str "%s" LOG_TIMESTAMP_VAL LOG_FLF_VAL "", ##__VA_ARGS__); \
                                                                                                         ^
C:/Users/andre/Desktop/development/llama.cpp/common/.\log.h:467:5: error: expected ')'
    LOG("01 Hello World to nobody, because logs are disabled!\n");
    ^
C:/Users/andre/Desktop/development/llama.cpp/common/.\log.h:300:27: note: expanded from macro 'LOG'
    #define LOG(str, ...) LOG_IMPL("%s" str, "", ##__VA_ARGS__, "")
                          ^
C:/Users/andre/Desktop/development/llama.cpp/common/.\log.h:251:102: note: expanded from macro 'LOG_IMPL'
            fprintf(LOG_TARGET, LOG_TIMESTAMP_FMT LOG_FLF_FMT str "%s" LOG_TIMESTAMP_VAL LOG_FLF_VAL "", ##__VA_ARGS__); \
                                                                                                     ^
C:/Users/andre/Desktop/development/llama.cpp/common/.\log.h:467:5: note: to match this '('
C:/Users/andre/Desktop/development/llama.cpp/common/.\log.h:300:27: note: expanded from macro 'LOG'
    #define LOG(str, ...) LOG_IMPL("%s" str, "", ##__VA_ARGS__, "")
                          ^
C:/Users/andre/Desktop/development/llama.cpp/common/.\log.h:251:20: note: expanded from macro 'LOG_IMPL'
            fprintf(LOG_TARGET, LOG_TIMESTAMP_FMT LOG_FLF_FMT str "%s" LOG_TIMESTAMP_VAL LOG_FLF_VAL "", ##__VA_ARGS__); \
                   ^
C:/Users/andre/Desktop/development/llama.cpp/common/.\log.h:300:50: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
    #define LOG(str, ...) LOG_IMPL("%s" str, "", ##__VA_ARGS__, "")
                                                 ^
C:/Users/andre/Desktop/development/llama.cpp/common/.\log.h:251:106: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
            fprintf(LOG_TARGET, LOG_TIMESTAMP_FMT LOG_FLF_FMT str "%s" LOG_TIMESTAMP_VAL LOG_FLF_VAL "", ##__VA_ARGS__); \
                                                                                                         ^
C:/Users/andre/Desktop/development/llama.cpp/common/.\log.h:469:5: error: expected ')'
    LOG("02 Hello World to default output, which is \"%s\" ( Yaaay, arguments! )!\n", LOG_STRINGIZE(LOG_TARGET));
```

I compiled that file with the -E flag to get the preprocessed output. This is where the error occurs:

```
do { if (log_handler() != nullptr) { fprintf(log_handler(), "[%" "llu" "] " "[%24s:%5ld][%24s] " "%s" "01 Hello World to nobody, because logs are disabled!\n" "%s" , (std::chrono::duration_cast<std::chrono::duration<std::uint64_t>>(std::chrono::system_clock::now().time_since_epoch())).count() , "C:/Users/andre/Desktop/development/llama.cpp/common/.\\log.h", 467, __FUNCTION__ "", "", ""); fflush(log_handler()); } } while (0);
```

And tried compiling the preprocessed output:

```
C:/Users/andre/Desktop/development/llama.cpp/common/.\log.h:467:383: error: expected ')'
  ..., "C:/Users/andre/Desktop/development/llama.cpp/common/.\\log.h", 467, __FUNCTION__ "", "", ""); fflush(log_hand...
                                                                                         ^
C:/Users/andre/Desktop/development/llama.cpp/common/.\log.h:467:49: note: to match this '('
    do { if (log_handler() != nullptr) { fprintf(log_handler(), "[%" "llu" "] " "[%24s:%5ld][%24s] " "%s" "01 He...
```

--- a/common/log.h
+++ b/common/log.h
@@ -234,7 +234,7 @@ inline std::string log_filename_generator_impl(LogTriState multilog, const std::
 // INTERNAL, DO NOT USE
 //  USE LOG() INSTEAD
 //
-#if !defined(_MSC_VER) || defined(__INTEL_LLVM_COMPILER)
+#if !defined(_MSC_VER) || defined(__INTEL_LLVM_COMPILER) || defined(__clang__)
     #define LOG_IMPL(str, ...)                                                                                      \
     do {                                                                                                            \
         if (LOG_TARGET != nullptr)                                                                                  \
@@ -257,7 +257,7 @@ inline std::string log_filename_generator_impl(LogTriState multilog, const std::
 // INTERNAL, DO NOT USE
 //  USE LOG_TEE() INSTEAD
 //
-#if !defined(_MSC_VER) || defined(__INTEL_LLVM_COMPILER)
+#if !defined(_MSC_VER) || defined(__INTEL_LLVM_COMPILER) || defined(__clang__)
     #define LOG_TEE_IMPL(str, ...)                                                                                                      \
     do {                                                                                                                                \
         if (LOG_TARGET != nullptr)                                                                                                      \

